lib/core/logger.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <iostream>
+#include <optional>
+#include <string>
+
+namespace hyped::utils {
+
+enum class Level { kNone, kDebug, kInfo, kWarn, kFatal };
+
+class ILogger {
+ public:
+  ILogger(const char *const module, const Level level) : module_(module), level_(level){};
```

Comment:
Do we really want constructors here?

Reply:
I think i did this because module_ and level_ are members of ILogger and not Logger so to set them i'd need to defer to ILoggers constructor?

Reply:
Changed the location of module_ and level_ to Logger instead of ILogger